common/src/main/java/com/wynntils/models/items/encoding/impl/block/DamageDataTransformer.java

```diff
@@ -0,0 +1,128 @@
+import com.wynntils.utils.type.ArrayReader;
+import com.wynntils.utils.type.ErrorOr;
+import com.wynntils.utils.type.Pair;
+import com.wynntils.utils.type.RangedValue;
+import com.wynntils.utils.type.UnsignedByte;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class DamageDataTransformer extends DataTransformer<DamageData> {
+    @Override
+    protected ErrorOr<UnsignedByte[]> encodeData(ItemTransformingVersion version, DamageData data) {
+        return switch (version) {
+            case VERSION_1 -> encodeDamageData(data);
+        };
+    }
+
+    @Override
+    public ErrorOr<DamageData> decodeData(ItemTransformingVersion version, ArrayReader<UnsignedByte> byteReader) {
+        return switch (version) {
+            case VERSION_1 -> decodeDamageData(byteReader);
+        };
+    }
+
+    @Override
+    public byte getId() {
+        return 10;
+    }
+
+    @Override
+    protected boolean shouldEncodeData(ItemTransformingVersion version, DamageData data) {
+        return !data.damages().isEmpty() || data.attackSpeed().isPresent();
+    }
+
+    private ErrorOr<UnsignedByte[]> encodeDamageData(DamageData data) {
+        List<UnsignedByte> bytes = new ArrayList<>();
+
+        if (data.attackSpeed().isEmpty()) {
+            return ErrorOr.error("Attack speed is not present, but damage data is present.");
+        }
+
+        // The first byte is the id of the attack speed of the item.
+        bytes.add(UnsignedByte.of((byte) data.attackSpeed().get().ordinal()));
+
+        // The next byte is the number of attack damages present on the item.
+        bytes.add(UnsignedByte.of((byte) data.damages().size()));
+
+        // An attack damage is encoded the following way:
+        for (Pair<DamageType, RangedValue> damage : data.damages()) {
+            // The first byte is the id of the skill (`ETFWAN`, where N represents Neutral).
+            DamageType damageType = damage.a();
+            // Neutral is 5
+            byte damageTypeId = 5;
+            if (damageType != DamageType.NEUTRAL && damageType.getElement().isEmpty()) {
+                return ErrorOr.error("Damage type " + damageType + " does not have an element");
+            } else if (damageType != DamageType.NEUTRAL) {
+                damageTypeId = (byte) damageType.getElement().get().ordinal();
+            }
+            bytes.add(UnsignedByte.of(damageTypeId));
+
+            // The next byte is the length of the integer that the minimum damage can fit in.
+            // The next bytes are the minimum damage bytes, which are assembled into an integer.
+            UnsignedByte[] unsignedBytes =
+                    UnsignedByteUtils.encodeVariableSizedInteger(damage.b().low());
+            bytes.add(UnsignedByte.of((byte) unsignedBytes.length));
+            bytes.addAll(List.of(unsignedBytes));
+
+            // The next byte is the length of the integer that the maximum damage can fit in.
```

Comment:
Just a reflection, maybe it is too late to change it now, and maybe it really doesn't matter, but there exists more compact representations for variable sized integers, that are assumed to be normally very low. I'm thinking of methods like UTF-8. Basically, the process is like: 
Store 7 bits of the source data in the target byte. Use the highest bit to mark if we're done. If it is 0, then we're done, and the byte is exactly the value we wanted. If it is 1, use the 7 bits we have, and grab the next 7 bits from the next byte. If that byte's highest bit is 0, then we're done (the value could be stored in 14 bits), otherwise continue.

Reply:
I've added it! It's actually great, since we encode a lot of numbers that fit 7 bits, so we save a fair amount of bytes. Implementation took a while, but I think it should be stable now :)